build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/snyk/SnykDependencyMonitoringGradlePluginFuncTest.groovy

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.internal.snyk
+
+import org.elasticsearch.gradle.fixtures.AbstractGradleInternalPluginFuncTest
+import org.elasticsearch.gradle.fixtures.http.HttpServerRule
+import org.elasticsearch.gradle.internal.conventions.precommit.LicenseHeadersPrecommitPlugin
+import org.gradle.api.Plugin
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.Rule
+import org.skyscreamer.jsonassert.JSONAssert
+
+class SnykDependencyMonitoringGradlePluginFuncTest extends AbstractGradleInternalPluginFuncTest {
+
+    @Rule
+    public HttpServerRule httpServer = new HttpServerRule()
+
+    Class<? extends Plugin> pluginClassUnderTest = SnykDependencyMonitoringGradlePlugin.class
+
+    def setup() {
+        configurationCacheCompatible = false // configuration is not cc compliant
+    }
+
+    def "can calculate snyk dependency graph"() {
+        given:
+        buildFile << """
+            apply plugin:'java'
+            version = "1.0-SNAPSHOT"
+
+            repositories {
+                mavenCentral() 
+            }
+            
+            dependencies {
+                implementation 'org.apache.lucene:lucene-monitor:9.2.0'
```

Comment:
For now, I don't want to go down the route of creating a maven test fixture to reproduce dependency graphs from scratch. Therefore we test against real dependencies from real repositories.